include/bsp/iotc_bsp_io_net.h

```diff
@@ -142,167 +159,126 @@ typedef enum iotc_bsp_protocol_type_e {
+  /** <code>1</code> if an error request is made to a socket, <code>0</code>
    * otherwise. */
   uint8_t in_socket_want_error : 1;
-  /** <code>1</code> if socket waits for a connected, <code>0</code> otherwise.
-   */
+  /** <code>1</code> if a connection request is made to a socket, <code>0</code>
+   * otherwise. */
   uint8_t in_socket_want_connect : 1;
-  /** <code>1</code> if socket can read, <code>0</code> otherwise. */
+  /** <code>1</code> if data can be read from a socket, <code>0</code>
+   * otherwise. */
   uint8_t out_socket_can_read : 1;
-  /** <code>1</code> if socket can write, <code>0</code> otherwise. */
+  /** <code>1</code> if data can be written to a socket, <code>0</code>
+   * otherwise. */
   uint8_t out_socket_can_write : 1;
-  /** <code>1</code> if an error occurs on the socket, <code>0</code> otherwise
-   */
+  /** <code>1</code> if an error occurs on the socket, <code>0</code>
+   * otherwise. */
   uint8_t out_socket_error : 1;
-  /** <code>1</code> if the connection process is finished, <code>0</code>
-   * otherwise */
+  /** <code>1</code> if a connection request succeeded, <code>0</code>
+   * otherwise. */
   uint8_t out_socket_connect_finished : 1;
 } iotc_bsp_socket_events_t;
 
 /**
- * @function
- * @brief Create a socket and connect it to an endpoint.
+ * @details Creates a socket and connects it to an endpoint.
  *
- * @param [out] iotc_socket The platform-specific socket representation should
- *     be stored in this parameter.  This value will be passed to all further
- *     networking bsp calls.
+ * @param [out] iotc_socket The platform-specific socket representation
+ *     This value is passed to all further BSP networking calls.
  * @param [in] host The null-terminated IP or fully-qualified domain name of the
- *     host to connect to.
+ *     host at which to connect.
  * @param [in] port The port number of the endpoint.
- * @param [in] socket_type The socket type. Can be <code>TCP</code>
- *     or <code>UDP</code>.
- *
- * @retval IOTC_BSP_IO_NET_STATE_OK Socket successfully connected to host.
- * @retval IOTC_BSP_IO_NET_STATE_ERROR Socket didn't connect.
+ * @param [in] socket_type The {@link #iotc_bsp_socket_type_e socket protocol}.
  */
 iotc_bsp_io_net_state_t iotc_bsp_io_net_socket_connect(
     iotc_bsp_socket_t* iotc_socket, const char* host, uint16_t port,
     iotc_bsp_socket_type_t socket_type);
 
 /**
- * @function
- * @brief Query socket states to schedule read and write operations.
+ * @brief Checks a {@link iotc_bsp_io_net_socket_connect() socket} for scheduled
+ * read or write operations.
  *
- * In the implementation of this function, each element in the
- * socket_events_array parameter must correspond to a socket and contain an
- * initialized iotc_bsp_socket_state_t structure.
- *
- * @param [in] socket_events_array An array of socket events.
+ * @param [in] socket_events_array An array of initialized sockets.
```

Comment:
These are really socket events and not sockets. Initialized, hm, it isn't really necessary here. Just fall back to the previous version.

Reply:
Done. Thanks - looks like I misinterpreted the phrase "socket events"